src/ducc0/infra/threading.cc

```diff
@@ -648,7 +648,7 @@ void Distribution::thread_map(std::function<void(Scheduler &)> f)
   auto new_f = YCombinator([this, &f, &counter, &ex, &ex_mut, pool](auto &new_f, size_t istart, size_t step) -> void {
     try
       {
-      ScopedValueChanger changer(in_parallel_region, true);
+      ScopedValueChanger<bool> changer(in_parallel_region, true);
```

Comment:
Actually, this breaks when using custom threading because `in_parallel_region` isn't defined (unless I actually add it to my custom threading modules).  Previously it was only used in the default ducc threadpool.

The pool's worker threads already set this value to true, so is this needed here?

Reply:
In this location it's probably not required,but in the one below we probably can't avoid it (sorry, I noticed this problem only very late...)

Would it be a better solution to have something like
```C++
try {
   auto current_parallelism = get_and_set_parallelism(false);
   MyScheduler sched(*this, 0);
   f(sched);
   get_and_set_parallelism(current_parallelism);
}
``` 
?

You'd still have to add a function (probably empty), but this approach is perhaps more conventional.

Reply:
I see, the last one is needed because the calling thread may not be a thread-pool worker.

> auto current_parallelism = get_and_set_parallelism(false);

Hmm... it's not pretty.  I guess you would have another class like `ScopedDisableParallelism`.

Are you sure you want to always disable parallelism when executing subtasks?  The `Distribution` isn't necessarily configured to use all available threads in the thread-pool.  In generic divide-and-conquer tasks, you may want to continue spinning off new multithreaded subtasks.

The other approach would be to force the caller (or `f`) to handle this, rather than the thread-pool/distribution.  The subtask itself would need to set _its_ `nthreads` explicitly for nested tasks.  In the case of FFT, I guess you would want to check the value returned by `thread_count(...)` to determine whether or not to further parallelize nested tasks.

Reply:
I'll think about this,but unfortunately I'll be AFK for the next two weeks. If you'd like me to add a tag (or a branch) before this change, so you can continue with your own work for the time being, let me know!

Reply:
> I'll be AFK for the next two weeks. 

Np.  Regarding timing, we have an intern, @RuijiaoSun, who will be working on `fftn` for TensorFlow during the last two weeks of August, so I'm trying to get things into TF and XLA by then.  I'm hitting a few more roadblocks when compiling for mobile, so I might try to send a few more patches for discussion over the next few weeks, but no rush.

For now I'll just create an `in_parallel_region` thread-local variable for ourselves as well.  I think we do still need the CTAD fix though.

Reply:
Sure!

Reply:
Nice to have a chance to work with you! Just want to correct the time, I will have the first two weeks in August to work on fftn. ^_^

Reply:
Great! I should be back on August 2.